### PR TITLE
Add a lifecycle test for GCE

### PR DESCRIPTION
**Pull request by kenji-cloudnatix**
- Move MockGCECloud to cloudmock/gce.
- Change Compute() and CloudDNS() of GCECloud to return interfaces
  for mocking

### CODE REVIEW

File: pkg/resources/gce/gce.go [hunk 1]
```diff
@@ -197,39 +197,37 @@ func (d *clusterDiscoveryGCE) listInstanceGroupManagersAndInstances() ([]*resour
...
+		is, err := c.Compute().InstanceGroupManagers().List(ctx, project, zoneName)
+		if err != nil {
+			return nil, fmt.Errorf("error listing InstanceGroupManagers: %v", err)
+		}
+		for i := range is {
+			mig := is[i] // avoid closure-in-loop go-tcha
+			instanceTemplate := instanceTemplates[mig.InstanceTemplate]
+			if instanceTemplate == nil {
+				klog.V(2).Infof("Ignoring MIG with unmanaged InstanceTemplate: %s", mig.InstanceTemplate)
+				klog.Infof("Ignoring MIG with unmanaged InstanceTemplate: %s", mig.InstanceTemplate)
```
Comment (rifelpet) [inline pkg/resources/gce/gce.go:209]:
```suggestion
```

**Review by rifelpet — COMMENTED** [review]
looks great, thanks for implementing this!

one nit otherwise lgtm. We can address the TODOs in followup PRs

**Comment by kenji-cloudnatix** [comment]
Thanks @hakman , @rifelpet ! Made a fix and rebased the PR.

**Comment by hakman** [comment]
Thanks @kenji-cloudnatix! :)

**Review by hakman — APPROVED** [review]

**Pull request merged**